src/python/pants/backend/python/rules/python_test_runner.py

```diff
@@ -22,6 +24,26 @@
 from pants.rules.core.strip_source_root import SourceRootStrippedSources
 
 
+def calculate_timeout_seconds(
+  *,
+  timeouts: bool,
+  test_target_timeout_seconds: Optional[int],
+  timeout_default_seconds: Optional[int],
+  timeout_maximum_seconds: Optional[int],
+) -> Optional[int]:
+  """Calculate the timeout for a test target.
+
+  If a target has no timeout configured its timeout will be set to the default timeout.
+  """
+  if not timeouts:
+    return None
+  if test_target_timeout_seconds is None:
+    test_target_timeout_seconds = timeout_default_seconds
+  if timeout_maximum_seconds is not None and test_target_timeout_seconds is not None:
+    return min(test_target_timeout_seconds, timeout_maximum_seconds)
+  return test_target_timeout_seconds
```

Comment:
I think this may express the logic even more clearly:

```python
if test_target_timeout_seconds is None:
  if timeout_default_seconds is None:
    return None
  test_target_timeout_seconds = timeout_default
if timeout_maximum_seconds is not None:
  return min(test_target_timeout_seconds, timeout_maximum_seconds)
return test_target_timeout_seconds
```

Or

```python
if test_target_timeout_seconds is None:
  if timeout_default_seconds is None:
    return None
  test_target_timeout_seconds = timeout_default
return min(test_target_timeout_seconds, timeout_maximum_seconds or -1)
```